target/linux/ramips/dts/mt7620a_dlink_dwr-961-a1.dts

```diff
@@ -0,0 +1,241 @@
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/mtd/partitions/uimage.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	compatible = "dlink,dwr-961-a1", "ralink,mt7620a-soc";
+	model = "D-Link DWR-961 A1";
+
+	aliases {
+		led-boot = &led_status;
+		led-failsafe = &led_status;
+		led-running = &led_status;
+		led-upgrade = &led_status;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status: led-0 {
+			function = LED_FUNCTION_STATUS;
+			color = <LED_COLOR_ID_GREEN>;
+			label = "green:status";
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+		};
+
+		led-1 {
+			function = LED_FUNCTION_WAN;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+		};
+
+		led-2 {
+			function = LED_FUNCTION_LAN;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+
+		led-3 { /* hidden next to wlan5g led */
+			function = LED_FUNCTION_DEBUG;
+			function-enumerator = <1>;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+		};
+
+		led-4 { /* hidden next to sms led*/
+			function = LED_FUNCTION_DEBUG;
+			function-enumerator = <2>;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+		};
+
+		led-5 {
+			function = LED_FUNCTION_SMS;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
+		};
+
+		led-6 {
+			function = LED_FUNCTION_SIGNAL;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		led-7 {
+			function = LED_FUNCTION_SIGNAL;
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&gpio2 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		led-8 {
+			function = LED_FUNCTION_CONNECTION;
+			function-enumerator = <4>;
```

Comment:
The `function-enumerator` of connection/wlan LEDs are used to denote the gen/band, right?

Reply:
Yes. 4 is from WIFI4.

Reply:
Hmm, I feel this kind of `function-enumerator` usage is a hack (though I like it)... Upstream kernel uses it only to denote the LED order.
https://github.com/torvalds/linux/blob/53ab78cd6d5aba25575a7cfb95729336ba9497d8/Documentation/devicetree/bindings/leds/common.yaml#L51

Reply:
Yeah, it's little hack. I can change it, but I like this solution too.